### PR TITLE
[Service Bus] Added java doc documentation on builder

### PR DESCRIPTION
**Pull request by hemanttanwar**
As identifies in past API review, Adding java doc for builder

-- Adding java doc on `ServiceBusClientBuilder`
-- Adding connection sharing in Readme.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 1]
```diff
@@ -56,6 +56,24 @@
 /**
  * The builder to create Service Bus clients:
  *
+ * <p><strong>Instantiate sync receiver</strong></p>
+ * {@codesnippet com.azure.messaging.servicebus.receiver.sync.client.instantiation}
+ *
+ * <p><strong>Instantiate sync sender</strong></p>
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:62]: I think showing how to instantiate one of each type of client is enough.
* Receiver
* Sender
* SessionReceiver
* Processor

Having an exhaustive matrix adds more cognitive load.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 2]
```diff
@@ -56,6 +56,24 @@
 /**
  * The builder to create Service Bus clients:
  *
+ * <p><strong>Instantiate sync receiver</strong></p>
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:59]: Instantiate a synchronous receiver

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 3]
```diff
@@ -56,6 +56,24 @@
...
+ * <p><strong>Instantiate sync sender</strong></p>
+ * {@codesnippet com.azure.messaging.servicebus.sender.sync.client.instantiation}
+ *
+ * <p><strong>Instantiate async receiver</strong></p>
+ * {@codesnippet com.azure.messaging.servicebus.receiver.async.client.instantiation}
+ *
+ * <p><strong>Instantiate async sender</strong></p>
+ * {@codesnippet com.azure.messaging.servicebus.sender.async.client.instantiation}
+ *
+ * <p><strong>Connection Sharing</strong></p>
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:71]: * "Sharing a connection between clients"
* Use title casing

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java [hunk 4]
```diff
@@ -59,17 +74,20 @@ public void instantiateSenderSync() {
...
+        // Retrieve 'connectionString' and 'queueName' from your configuration.
         ServiceBusClientBuilder builder = new ServiceBusClientBuilder()
             .connectionString(connectionString);
-        ServiceBusSenderAsyncClient sender = builder
-            .sender()
-            .topicName(topicName)
-            .buildAsyncClient();
-        // END: com.azure.messaging.servicebus.sender.async.client.instantiation
-        sender.sendMessage(new ServiceBusMessage("payload")).subscribe();
+        ServiceBusProcessorClient processor = builder
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java:82]: The builder methods can be chained instead.
```java
ServiceBusProcessorClient processor = new ServiceBusClientBuilder()
            .connectionString(connectionString)
            .processor()
            .queueName(queueName)
            .processMessage(System.out::println)
            .processError(context -> System.err.println(context.getErrorSource()))
            .buildProcessorClient();

```

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java [hunk 5]
```diff
@@ -44,6 +33,32 @@ public void instantiateReceiverAsync() {
...
+            .topicName(topicName)
+            .subscriptionName(subscriptionName)
+            .buildAsyncClient();
+
+        // Receiving messages from the first available sessions. It waits up to the AmqpRetryOptions.getTryTimeout().
+        // If no session is available within that operation timeout, it completes with an error. Otherwise, a receiver
+        // is returned when a lock on the session is acquired.
+        Mono<ServiceBusReceiverAsyncClient> receiverMono = sessionReceiver.acceptNextSession();
+
+        Disposable subscription = Flux.usingWhen(receiverMono,
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java:54]: Why do we assign this to a subscription? ` subscription.dispose();` is not part of the snippet so, the assignment may not be required.

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java [hunk 6]
```diff
@@ -44,6 +33,32 @@ public void instantiateReceiverAsync() {
         receiver.receiveMessages().blockFirst(Duration.ofSeconds(1));
     }
 
+    public void instantiateSessionReceiver() {
+        // BEGIN: com.azure.messaging.servicebus.session.receiver.async.client.instantiation
+        // Retrieve 'connectionString', 'topicName' and 'subscriptionName' from your configuration.
+        ServiceBusClientBuilder builder = new ServiceBusClientBuilder()
+            .connectionString(connectionString);
+        // Create a session receiver.
+        ServiceBusSessionReceiverAsyncClient sessionReceiver = builder
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java:42]: The builder methods should be chained to demonstrate a better pattern when not sharing connections.

**Review by srnagar — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java (hunk 4 above)
Comment (hemanttanwar) [reply sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusClientBuilderJavaDocCodeSamples.java:82]: In intentionally kept it this way , so it promotes sharing of connection.

**Pull request merged**